### DISCUSSION TITLE
R9 Slim not plus

### DISCUSSION THREAD

**Discussion by Christianmix** [ExpressLRS/ExpressLRS#768] [Q&A]
I'm trying to install firmware on my slim r9 but I'm having problems, does anyone have any idea what might be going on?

log

Processing Frsky_RX_R9SLIM_via_BetaflightPassthrough (platform: ststm32@8.0.0; board: bluepill_f103c8; framework: arduino)
--------------------------------------------------------------------------------
Verbose mode can be enabled via `-v, --verbose` option
UID bytes: 66,30,193,21,197,23
build flags: ['-D PLATFORM_STM32=1', '-O2', '-DTARGET_RX=1 -Wall -Iinclude', '-D TARGET_R9M_RX', '-D TARGET_R9SLIM_RX', '-D TARGET_100mW_MODULE', '-D HSE_VALUE=12000000U', '-DVECT_TAB_OFFSET=0x08008000U', '-DRegulatory_Domain_FCC_915', '-DMY_UID=66,30,193,21,197,23', '-DHYBRID_SWITCHES_8', '-DLOCK_ON_FIRST_CONNECTION', '-DFAST_SYNC', '-DLATEST_COMMIT=0x1,0xd,0x5,0x4,0x3,0xb']
Building for STM32 Platform
Building for SX1276 915FCC
CONFIGURATION: https://docs.platformio.org/page/boards/ststm32/bluepill_f103c8.html
PLATFORM: ST STM32 (8.0.0) > BluePill F103C8
HARDWARE: STM32F103C8T6 72MHz, 20KB RAM, 64KB Flash
DEBUG: Current (stlink) External (blackmagic, jlink, stlink)
PACKAGES: 
 - framework-arduinoststm32 4.10900.200819 (1.9.0) 
 - framework-cmsis 2.50501.200527 (5.5.1) 
 - tool-dfuutil 1.9.200310 
 - tool-openocd 2.1000.200630 (10.0) 
 - tool-stm32duino 1.0.2 
 - toolchain-gccarmnoneeabi 1.90201.191206 (9.2.1)
LDF: Library Dependency Finder -> http://bit.ly/configure-pio-ldf
LDF Modes: Finder ~ chain, Compatibility ~ soft
Found 34 compatible libraries
Scanning dependencies...
Dependency Graph
|-- <extEEPROM> 3.4.1+sha.4abc28d
|   |-- <Wire> 1.0
|-- <Wire> 1.0
|-- <SX127xDriver>
|   |-- <SPI> 1.0
|-- <SX1280Driver>
|   |-- <SPI> 1.0
|-- <FHSS>
|   |-- <SX127xDriver>
|   |   |-- <SPI> 1.0
|   |-- <SX1280Driver>
|   |   |-- <SPI> 1.0
|-- <POWERMGNT>
|   |-- <DAC>
|   |   |-- <Wire> 1.0
|   |   |-- <helpers>
|   |-- <SX127xDriver>
|   |   |-- <SPI> 1.0
|   |-- <SX1280Driver>
|   |   |-- <SPI> 1.0
|-- <EEPROM>
|   |-- <EEPROM> 2.0.1
|   |-- <extEEPROM> 3.4.1+sha.4abc28d
|   |   |-- <Wire> 1.0
|   |-- <Wire> 1.0
|-- <CRC>
|-- <CRSF>
|   |-- <CRC>
|   |-- <CrsfProtocol>
|   |-- <MSP>
|   |-- <TelemetryProtocol>
|   |-- <FIFO>
|-- <HWTIMER>
|-- <LQCALC>
|-- <MSP>
|-- <OTA>
|   |-- <CRSF>
|   |   |-- <CRC>
|   |   |-- <CrsfProtocol>
|   |   |-- <MSP>
|   |   |-- <TelemetryProtocol>
|   |   |-- <FIFO>
|-- <PFD>
|-- <STM32F3_WS2812B>
|-- <StubbornReceiver>
|-- <StubbornSender>
|-- <Telemetry>
|   |-- <CrsfProtocol>
|-- <TelemetryProtocol>
Building in release mode
PLATFORM : 'ststm32'
BUILD ENV: 'FRSKY_RX_R9SLIM_VIA_BETAFLIGHTPASSTHROUGH'
Checking size .pio\build\Frsky_RX_R9SLIM_via_BetaflightPassthrough\firmware.elf
Advanced Memory Usage is available via "PlatformIO Home > Project Inspect"
RAM:   [==        ]  21.1% (used 4328 bytes from 20480 bytes)
Flash: [======    ]  56.1% (used 36748 bytes from 65536 bytes)
Configuring upload protocol...
AVAILABLE: blackmagic, custom, dfu, jlink, mbed, stlink
CURRENT: upload_protocol = custom
Uploading .pio\build\Frsky_RX_R9SLIM_via_BetaflightPassthrough\firmware.bin
=================== FIRMWARE UPLOAD ===================
  Bin file '.pio\build\Frsky_RX_R9SLIM_via_BetaflightPassthrough\firmware.bin'
  Port COM3 @ 420000
  Using CRSF (full duplex)!
======== PASSTHROUGH INIT ========
  Trying to initialize COM3 @ 420000
  Warning: 'No CLI available. Already in passthrough mode?, If this fails reboot FC and try again!'

Attempting to reboot into bootloader...
[1] retry...
[2] retry...
[3] retry...
[4] retry...
[5] retry...
[6] retry...
[7] retry...
[8] retry...
[9] retry...
[10] retry...
[FAILED] to get to BL in reasonable time
[FAILED] to get to BL in reasonable time

*** [upload] Error -1
========================= [FAILED] Took 30.90 seconds =========================

Environment                                Status    Duration
-----------------------------------------  --------  ------------
Frsky_RX_R9SLIM_via_BetaflightPassthrough  FAILED    00:00:30.900
==================== 1 failed, 0 succeeded in 00:00:30.900 ====================

Flash error
*** [upload] Error -1 ========================= [FAILED] Took 29.36 seconds ========================= Environment Status Duration ----------------------------------------- -------- ------------ Frsky_RX_R9SLIM_via_BetaflightPassthrough FAILED 00:00:29.357 ==================== 1 failed, 0 succeeded in 00:00:29.357 ====================

**Comment by sandrosov85** [comment]
1st stage - write bootloader in r9slim (via radio transmitter).
2nd stage - boot r9slim with pressed button in bootloader mode and connected to your FC.
3rd stage - write fw to r9slim via FC.

I do that steps with my 2 r9slim+ modules and it works just fine...

upd. elsr documentation about fw update on r9 series receivers provided [here](https://www.expresslrs.org/1.1/quick-start/rx-r9receivers/)
